gateway/server.go

```diff
@@ -33,67 +28,32 @@ func main() {
 	log.Printf("HTTP Read Timeout: %s", config.ReadTimeout)
 	log.Printf("HTTP Write Timeout: %s", config.WriteTimeout)
 
-	var dockerClient *client.Client
-
-	if config.UseExternalProvider() {
-		log.Printf("Binding to external function provider: %s", config.FunctionsProviderURL)
-	} else {
-		var err error
-		dockerClient, err = client.NewEnvClient()
-		if err != nil {
-			log.Fatal("Error with Docker client.")
-		}
-		dockerVersion, err := dockerClient.ServerVersion(context.Background())
-		if err != nil {
-			log.Fatal("Error with Docker server.\n", err)
-		}
-		log.Printf("Docker API version: %s, %s\n", dockerVersion.APIVersion, dockerVersion.Version)
+	if !config.UseExternalProvider() {
+		log.Fatalln("As of this version of OpenFaaS, you must use external provider even for Docker Swarm.")
```

Comment:
This error message will only be relevant to users who are upgrading from the previous version and won't make much sense to new users (or in the future)
I'd suggest something along the lines of "You must define an orchestration provider. Swarm and Kubernetes are supported"